src/ngx_http_lua_string.c

```diff
@@ -433,17 +433,20 @@ ngx_http_lua_ffi_unescape_uri(const u_char *src, size_t len, u_char *dst)
 
 
 size_t
-ngx_http_lua_ffi_uri_escaped_length(const u_char *src, size_t len)
+ngx_http_lua_ffi_uri_escaped_length(const u_char *src, size_t len,
+    int not_component)
 {
-    return len + 2 * ngx_http_lua_escape_uri(NULL, (u_char *) src, len,
-                                             NGX_ESCAPE_URI_COMPONENT);
+    int type = not_component ? NGX_ESCAPE_URI :  NGX_ESCAPE_URI_COMPONENT;
```

Comment:
Style: seems like you use two spaces after the colon (`:`)?

I don't think you need to introduce the `type` variable in the first place.